src/target/riscv/riscv-013.c

```diff
@@ -1187,23 +1185,44 @@ static int examine(struct target *target)
 
 static int assert_reset(struct target *target)
 {
-	/*FIXME -- this only works for single-hart.*/
 	RISCV_INFO(r);
-	assert(r->current_hartid == 0);
 
 	select_dmi(target);
-	LOG_DEBUG("ASSERTING NDRESET");
-	uint32_t control = dmi_read(target, DMI_DMCONTROL);
-	control = set_field(control, DMI_DMCONTROL_NDMRESET, 1);
-	if (target->reset_halt) {
-		LOG_DEBUG("TARGET RESET HALT SET, ensuring halt is set during reset.");
-		control = set_field(control, DMI_DMCONTROL_HALTREQ, 1);
+
+	uint32_t control_base = set_field(0, DMI_DMCONTROL_DMACTIVE, 1);
+
+	if (target->rtos) {
+		// There's only one target, and OpenOCD thinks each hart is a thread.
+		// We must reset them all.
```

Comment:
Maybe should update this comment to say "we must reset them all, but we only halt the ones controlled by this target"

Reply:
There is only one target in the RTOS case, and that target controls all the harts.